scripts/add-validator/index.ts

```diff
@@ -41,38 +47,172 @@ class AddValidator {
+    const bondStashAccount = await validator.bondValue(
+      this.api,
+      this.controllerAccount.address,
+      this.stashAccount,
+      bondValue
+    );
 
     // set session key
-    const setSessionKey = await validator.setSessionKey(this.api, sessionKey, this.controllerAccount);
+    const setSessionKey = await validator.setSessionKey(
+      this.api,
+      sessionKey,
+      this.controllerAccount
+    );
 
     // set reward commission
-    const reward = await validator.setCommission(this.api, rewardCommission, this.controllerAccount);
+    const reward = await validator.setCommission(
+      this.api,
+      rewardCommission,
+      this.controllerAccount
+    );
+  }
+
+  public async createAccounts() {
+    console.log("Creating Stash and Controller accounts");
+    if (process.env.NETWORK === "mainnet") {
+      throw new Error("Couldn't create new account for Mainnet");
+    }
+    this.stashAccount = this.generateAccount("Stash");
+    this.controllerAccount = this.generateAccount("Controller");
+
+    const stashAssetsResponse = await this.requestAssets(this.stashAccount);
+    console.log("Stash assets transaction:", stashAssetsResponse?.data);
+
+    const controllerAssetsResponse = await this.requestAssets(
+      this.controllerAccount
+    );
+    console.log(
+      "Controller assets transaction:",
+      controllerAssetsResponse?.data
+    );
+
+    await this.callWithRetry(this.isValidBalance.bind(this));
+
+    const validator = new Validator();
+    // Fetch account balance
+    const { stashBalance, controllerBalance } = await validator.accountBalance(
+      this.api,
+      this.stashAccount.address,
+      this.controllerAccount.address
+    );
+    console.log(
+      `The stash account ${this.stashAccount.address} balance is ${stashBalance}`
+    );
+    console.log(
+      `The controller account ${this.controllerAccount.address} balance is ${controllerBalance}`
+    );
+  }
+
+  private generateAccount(type: string) {
+    const keyring = new Keyring({ type: "sr25519", ss58Format: 2 });
+    const mnemonic = mnemonicGenerate(12);
+    const pair = keyring.addFromUri(mnemonic, {}, "ed25519");
+
+    console.log("=====================================================");
+    console.log(`GENERATED 12-WORD MNEMONIC SEED (${type}):`);
+    console.log(mnemonic);
+    console.log("=====================================================");
+
+    return keyring.addPair(pair);
+  }
+
+  private async requestAssets(account: KeyringPair) {
+    try {
+      const network = process.env.NETWORK;
+      console.log(network);
+      console.log(network.toUpperCase());
+      return await axios.post(
+        process.env.REQUEST_ASSETS_ENDPOINT,
+        { destination: account.address, network: network.toUpperCase() },
+        {
+          timeout: 15000,
+          withCredentials: false,
+          headers: {
+            Accept: "application/json",
+            Authority: "laboratory-api.cere.network",
```

Comment:
Why do we need this header?

Reply:
It was there in our previous commit. I thought we need this to authenticate our API request.

Reply:
Check and confirm. If no, remove.

Reply:
Removed